docs/LlpcOverview.md

```diff
@@ -0,0 +1,71 @@
+# LLPC overview
+
+This document outlines the state of LLPC as it will be at the end of the
+"middle-end interface clean-up" series of patches.
+
+## Compilation process
+
+The LLPC compilation process consists of three phases:
+* front-end;
```

Comment:
Is it possible to add a data flow diagram to show the compilation or the top-level  constructs. I mean just like this

```
+----+
| FE |  ----->
+----+
```

Reply:
I made the diagram a bit more detailed than that. :-)